plugins/destination/clickhouse/client/migrate.go

```diff
@@ -52,27 +89,69 @@ func (c *Client) createTable(ctx context.Context, table *schema.Table) (err erro
 	return c.conn.Exec(ctx, queries.CreateTable(table))
 }
 
-func (c *Client) autoMigrateTable(ctx context.Context, current, table *schema.Table) (err error) {
-	tableName := table.Name
-	c.logger.Info().Str("table", tableName).Msg("Table exists, migrating")
+func (c *Client) dropTable(ctx context.Context, table *schema.Table) (err error) {
+	c.logger.Debug().Str("table", table.Name).Msg("Dropping table")
 
-	have, want := current.GetChangedColumns(table)
-	if len(have) > 0 && c.spec.MigrateMode != specs.MigrateModeForced {
-		return fmt.Errorf("table %s has different types for columns %v but schema wants %v , add `migrate_mode: forced` to the destination spec to drop the columns", table.Name, have, want)
+	return c.conn.Exec(ctx, queries.DropTable(table))
+}
+
+func needsTableDrop(change schema.TableColumnChange) bool {
+	// We can safely add a nullable column without dropping the table
+	if change.Type == schema.TableColumnChangeTypeAdd && !change.Current.CreationOptions.NotNull {
+		return false
 	}
 
-	for _, column := range want {
-		if err := c.conn.Exec(ctx, queries.DropColumn(tableName, &column)); err != nil {
-			return err
-		}
-		if err := c.conn.Exec(ctx, queries.AddColumn(tableName, &column)); err != nil {
-			return err
-		}
+	// We can safely drop a nullable and add a nullable column without dropping the table
+	if change.Type == schema.TableColumnChangeTypeUpdate && !change.Current.CreationOptions.NotNull {
+		return false
 	}
 
-	for _, column := range table.GetAddedColumns(current) {
-		if err := c.conn.Exec(ctx, queries.AddColumn(tableName, &column)); err != nil {
-			return err
+	// Not null columns are used as sort keys, so we need to drop the table if those are removed
+	if change.Type == schema.TableColumnChangeTypeRemove && change.Previous.CreationOptions.NotNull {
+		return false
+	}
+
+	return true
+}
+
+func changesSorter(a, b schema.TableColumnChange) bool {
+	aDropTable := needsTableDrop(a)
+	bDropTable := needsTableDrop(b)
+	return aDropTable && !bDropTable
+}
+
+func (c *Client) autoMigrate(ctx context.Context, table *schema.Table, current *schema.Table) (err error) {
+	changes := table.GetChanges(current)
+	sort.SliceStable(changes, func(i, j int) bool {
+		return changesSorter(changes[i], changes[j])
+	})
+	for _, change := range changes {
+		switch {
+		case change.Type == schema.TableColumnChangeTypeAdd && !change.Current.CreationOptions.NotNull:
+			c.logger.Debug().Str("table", table.Name).Str("column", change.Current.Name).Msg("Adding new column")
+			err := c.conn.Exec(ctx, queries.AddColumn(table.Name, &change.Current))
+			if err != nil {
+				return err
+			}
+		case change.Type == schema.TableColumnChangeTypeUpdate && !change.Current.CreationOptions.NotNull:
+			c.logger.Debug().Str("table", table.Name).Str("column", change.Current.Name).Msg("dropping changed non null column")
```

Comment:
```suggestion
			c.logger.Debug().Str("table", table.Name).Str("column", change.Current.Name).Msg("Recreating nullable column")
```

Reply:
This was removed as we now recreate a table on column changes due to not going with https://github.com/cloudquery/plugin-sdk/pull/697